pkg/commands/test/test.go

```diff
@@ -26,89 +25,123 @@ import (
 )
 
 var (
-	denyQ = regexp.MustCompile("^deny(_[a-zA-Z]+)*$")
-	warnQ = regexp.MustCompile("^warn(_[a-zA-Z]+)*$")
+	DenyQ                 = regexp.MustCompile("^deny(_[a-zA-Z]+)*$")
+	WarnQ                 = regexp.MustCompile("^warn(_[a-zA-Z]+)*$")
+	CombineConfigFlagName = "combine-config"
 )
 
-// checkResult describes the result of a conftest evaluation.
+// CheckResult describes the result of a conftest evaluation.
 // warning and failure "errors" produced by rego should be considered separate
 // from other classes of exceptions.
-type checkResult struct {
-	warnings []error
-	failures []error
+type CheckResult struct {
+	Warnings []error
+	Failures []error
 }
 
 // NewTestCommand creates a new test command
-func NewTestCommand() *cobra.Command {
+func NewTestCommand(osExit func(int), getOutputManager func() OutputManager) *cobra.Command {
 
 	ctx := context.Background()
 	cmd := &cobra.Command{
 		Use:     "test <file> [file...]",
 		Short:   "Test your configuration files using Open Policy Agent",
 		Version: fmt.Sprintf("Version: %s\nCommit: %s\nDate: %s\n", constants.Version, constants.Commit, constants.Date),
 
-		Run: func(cmd *cobra.Command, args []string) {
-
-			if len(args) < 1 {
+		Run: func(cmd *cobra.Command, fileList []string) {
+			out := getOutputManager()
+			if len(fileList) < 1 {
 				cmd.SilenceErrors = true
 				log.G(ctx).Fatal("The first argument should be a file")
 			}
 
 			if viper.GetBool("update") {
-				update.NewUpdateCommand().Run(cmd, args)
+				update.NewUpdateCommand().Run(cmd, fileList)
 			}
 
 			compiler, err := buildCompiler(viper.GetString("policy"))
 			if err != nil {
 				log.G(ctx).Fatalf("Problem building rego compiler: %s", err)
 			}
-
-			out := getOutputManager(viper.GetString("output"), !viper.GetBool("no-color"))
-
 			foundFailures := false
-			for _, fileName := range args {
+			var configFiles []parser.ConfigDoc
+			var fileType string
+			for _, fileName := range fileList {
+				var err error
+				var config io.ReadCloser
+
+				if fileName == "-" {
+					config = ioutil.NopCloser(bufio.NewReader(os.Stdin))
+					fileType = viper.GetString("input")
+				} else {
+					filePath, _ := filepath.Abs(fileName)
+					fileType = strings.TrimPrefix(filepath.Ext(fileName), ".")
```

Comment:
Force to use `input` flag even if the extension is valid(It may be the unknown but supported format). I.e command:
`./conftest test -p examples/ini/policy/ examples/ini/unknown.xyz -i ini`
`cat examples/traefik/traefik.toml | ./conftest test -i toml -p examples/traefik/policy/ -`

 				if viper.GetString("input") != "" {
					fileType = viper.GetString("input")
				} else {
					fileType = strings.TrimPrefix(filepath.Ext(fileName), ".")
				}
				if fileName == "-" {
					config = ioutil.NopCloser(bufio.NewReader(os.Stdin))
				} else {
					filePath, _ := filepath.Abs(fileName)
					config, err = os.Open(filePath)
					if err != nil {
						log.G(ctx).Fatalf("Unable to open file %s: %s", fileName, err)
					}
				}

Reply:
Docs to come in a later PR but now `stdin` requires the input flag. We opted for a declarative approach to avoid parsing errors and keep the user honest about what they were using.